projects/ngx-pwa/local-storage/src/lib/storages/storage-map.service.ts

```diff
@@ -162,14 +162,49 @@ export class StorageMap {
    *   }
    * });
    */
+  get<T extends string = string>(key: string, schema: JSONSchemaString): Observable<string | undefined>;
+  /**
+   * @deprecated The cast is useless here and doesn't match the JSON schema. Just remove the cast.
+   * @see {@link https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/VALIDATION.md}
+   */
   get<T = string>(key: string, schema: JSONSchemaString): Observable<string | undefined>;
+  get<T extends number = number>(key: string, schema: JSONSchemaInteger | JSONSchemaNumber): Observable<number | undefined>;
+  /**
+   * @deprecated The cast is useless here and doesn't match the JSON schema. Just remove the cast.
+   * @see {@link https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/VALIDATION.md}
+   */
   get<T = number>(key: string, schema: JSONSchemaInteger | JSONSchemaNumber): Observable<number | undefined>;
+  get<T extends boolean = boolean>(key: string, schema: JSONSchemaBoolean): Observable<boolean | undefined>;
+  /**
+   * @deprecated The cast is useless here and doesn't match the JSON schema. Just remove the cast.
+   * @see {@link https://github.com/cyrilletuzi/angular-async-local-storage/blob/master/docs/VALIDATION.md}
+   */
   get<T = boolean>(key: string, schema: JSONSchemaBoolean): Observable<boolean | undefined>;
+  get<T extends string[] = string[]>(key: string, schema: JSONSchemaArrayOf<JSONSchemaString>): Observable<string[] | undefined>;
```

Comment:
Wouldn't make sense to change this also to (the other cases also):

```suggestion
  get<T extends string[] = string[]>(key: string, schema: JSONSchemaArrayOf<JSONSchemaString>): Observable<T | undefined>;
```

If not, the generic parameter would be useless.

Reply:
Yes, you're right, thanks for pointing this out! Opened #469, can you check it's OK?